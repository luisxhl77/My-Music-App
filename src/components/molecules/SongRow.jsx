import styled from "styled-components";

export const SongRow = ({track}) => {
    console.log(track.length)
  return (
    <SongRowContainer>
        <img src={track.album.images[0].url} alt="album"/>
        <SongInfo>
            <h4>{track.name}</h4>
            <p>
                {
                    track.artists.map(artist => artist.name).join(', ')
                }
                {
                    track.album.name
                }
            </p>
        </SongInfo>
    </SongRowContainer>
  )
}

const SongRowContainer = styled.div`
    padding: 10px 0px;
    display: flex;
    margin-left: 25px;
    align-items: center;
    z-index: 90;
    color: #fff;
    &:hover{
        cursor: pointer;
        background-color: #000000;
        opacity: 0.8;
    }
    & img{
        height: 50px;
        width: 50px;
        object-fit: contain;
    }
`
const SongInfo = styled.div`
    
`