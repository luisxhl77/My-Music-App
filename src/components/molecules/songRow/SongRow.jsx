import { FavoriteBorder, PlayCircleFilled } from "@mui/icons-material";
import './songRow.scss';

export const SongRow = ({track, mySavedTracks}) => {
    return (
        <section className="songRow">
            <div className="songRow__info-music">
                <div className="songRow__play-music">
                    <figure className="songRow__image">
                        <img src={ (track) ? track?.album?.images[0]?.url : mySavedTracks?.album?.images[0]?.url} alt="album"/>
                        <span className="songRow__play">
                            <PlayCircleFilled fontSize="large"/>
                        </span>
                    </figure>
                </div>
                <div className="songRow__song-info">
                    <h4>{(track) ? track?.name : mySavedTracks?.name}</h4>
                    <p>
                        {(track) ? track.artists.map(artist => artist.name).join(', ') : mySavedTracks.artists.map(artist => artist.name).join(', ')}
                    </p>
                </div> 
            </div>
            <FavoriteBorder className="songRow__icon-favorite"/>
        </section>
    )
}