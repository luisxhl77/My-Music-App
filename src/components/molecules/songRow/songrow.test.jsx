import { render, screen } from "@testing-library/react"
import { SongRow } from "./SongRow"

describe('Prubas de <songRow/> ', () => {

    const id = '102837asd'; 
    const name = 'Wake Me Up'; 
    const image = 'https://avicii.com/avicii.jsg';
    const artist = 'Avicii';
    
    test('debe de hacer match con el snapshop', () => {
        const { container } = render(
            <SongRow
                id = {id}
                name = {name}
                image = {image}
                artist = {artist}
            />
        );

        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar el nombre de la cancion y el artista', () => {
        render(
            <SongRow
                id = {id}
                name = {name}
                image = {image}
                artist = {artist}
            />
        );

        expect( screen.getByText( name ) ).toBe( screen.getByLabelText('titleSong'));
        expect( screen.getByText( artist ) ).toBe( screen.getByLabelText('artisSong'));

    })

    test('debe de ser igual el src y el id con los argumentados mandados', () => {
        render(
            <SongRow
                id = {id}
                name = {name}
                image = {image}
                artist = {artist}
            />
        );

        const { src, alt } = screen.getByLabelText('imageSong');

        expect( src ).toBe( image );
        expect( alt ).toBe( id );

    })
    
})