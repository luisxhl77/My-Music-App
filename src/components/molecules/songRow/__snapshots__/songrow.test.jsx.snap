// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas de <songRow/>  debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="songRow"
  >
    <div
      class="songRow__info-music"
    >
      <div
        class="songRow__play-music"
      >
        <figure
          class="songRow__image"
        >
          <img
            alt="102837asd"
            aria-label="imageSong"
            src="https://avicii.com/avicii.jsg"
          />
          <span
            class="songRow__play"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="PlayCircleFilledIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
              />
            </svg>
          </span>
        </figure>
      </div>
      <div
        class="songRow__song-info"
      >
        <h4
          aria-label="titleSong"
          class="songRow__titleSong"
        >
           
          Wake Me Up
           
        </h4>
        <p
          aria-label="artisSong"
          class="songRow__description"
        >
           
          Avicii
           
        </p>
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium songRow__icon-favorite css-i4bv87-MuiSvgIcon-root"
      data-testid="FavoriteBorderIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
      />
    </svg>
  </div>
</div>
`;
